agent/consul/rate/handler.go

```diff
@@ -170,6 +175,10 @@ func NewHandler(cfg HandlerConfig, delegate HandlerDelegate, logger hclog.Logger
 // Note: this starts a goroutine.
 func (h *Handler) Run(ctx context.Context) {
 	h.limiter.Run(ctx)
+	discardLogger := hclog.NewInterceptLogger(&hclog.LoggerOptions{Output: io.Discard})
+	discardLogger.RegisterSink(logdrop.NewLogDropSink(ctx, 100, h.logger, func(l logdrop.Log) {
```

Comment:
just curious how do we select 100 as the depth?

Reply:
Circling back on this. The 100 is just a number that seemed fine in my head. I think we should refine that with more testing.
My logic is that 100 is enough to smoothen any peak and over than that it's fine to start dropping as most of those logs have duplicate information anyway.